server/src/test/java/io/crate/integrationtests/PartitionedTableConcurrentIntegrationTest.java

```diff
@@ -441,4 +443,101 @@ public void testConcurrentPartitionCreationWithColumnCreationTypeMissMatch() thr
         // this query should never fail
         execute("select * from t1 order by x limit 50");
     }
+
+    @Test
+    public void test_column_positions_after_concurrent_mapping_updates_on_partitioned_table() throws InterruptedException {
```

Comment:
Could we somehow unify this with the other one for the regular tables?

Maybe randomize the partition by clause or something like that?